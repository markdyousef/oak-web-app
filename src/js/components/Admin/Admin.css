:root {
    --primary-color: #34B289;
    --main-black-color: #131517;
    --light-black-color: #2d3438;
    --dark-gray-color: #737a80;
    --main-gray-color: #f2f2f2;
    --light-gray-color: #fafafa;
    --main-white-color: #fff;
}

.container {
    display: flex;
    flex-direction: column;
    align-items: center;
    height: 100vh;
    background: #34B289;
}

.teamList {
    height: 100%;
    position: absolute;
    top: 0;
    left: 380px;
    right: 0;
    bottom: 0;
}

.teamList::before {
    content: '';
    display: inline-block;
    vertical-align: middle;
    height: 100%;
}

.teamListInner {
    display: inline-block;
    vertical-align: middle;
    overflow: hidden;
    width: 100%;
    padding: 0px 80px;
    box-sizing: border-box;
    margin-top: -48px;
}

.loading {
    color: #fff;
    font-size: 16px;
}

.SidePanel {
    width: 380px;
    background: #fff;
    position: absolute;
    top: 0;
    left: 0;
    bottom: 0;
    z-index: 9999;
    & img {
        height: 28px;
        position: absolute;
        top: 16px;
        left: 20px;
    }
}

.SidePanelInner {
    height: 100%;
    display: flex;
    align-items: baseline;
    justify-content: center;
    flex-wrap: wrap;
    flex-grow: 0;
    padding: 0px 20px;
    flex-direction: column;
    text-align: left;
    margin-top: -16px;
    & h1 {
        font-size: 24px;
        font-weight: bold;
        color: #131517;
        display: block;
        padding: 0;
        text-align: left;
        margin-bottom: 16px;
    }
    & p {
        display: block;
        font-size: 15px;
        line-height: 1.4;
        max-width: 280px;
    }
}
