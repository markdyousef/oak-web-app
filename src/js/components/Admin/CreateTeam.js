// @flow
import React, { PropTypes } from 'react';
import Input from '../../components/shared/Input';
import Button from '../../components/shared/Button';

import css from './Admin.css';

const CreateTeam = ({ teamName, onChange, onSubmit }:Object) => {
    return (
        <div className={css.actionField}>
            <h1>Create a new team</h1>
            <div>
                <Input
                    value={teamName}
                    placeholder="Clai"
                    onChange={onChange}
                />
                <Button
                    onClick={onSubmit}
                    text="Create"
                    type="primary"
                />
            </div>
        </div>
    );
};

CreateTeam.propTypes = {
    teamName: PropTypes.string,
    onChange: PropTypes.func.isRequired,
    onSubmit: PropTypes.func.isRequired
};

CreateTeam.defaultProps = {
    teamName: null
};


export default CreateTeam;
