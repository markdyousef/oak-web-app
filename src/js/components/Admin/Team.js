// @flow
import React, { PropTypes } from 'react';

import css from './Admin.css';

const Team = ({ team, onSelect }) => {
    return (
        <div className={css.team}>
            <h4>{team.name}</h4>
        </div>
    );
};

Team.propTypes = {
    team: PropTypes.shape({
        id: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired
    }),
    onSelect: PropTypes.func.isRequired
};
