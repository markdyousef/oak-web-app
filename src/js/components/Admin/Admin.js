// @flow
import React, { Component } from 'react';
import styled from 'styled-components';
import CreateTeam from './CreateTeam';
import Team from './Team';
import logo from '../../../img/cuest-logo.svg';
import { selectTeam } from '../../utils';

import css from './Admin.css';

const Submit = styled.button `
    position: absolute;
    bottom: 80px;
    right: 80px;
    background: #fff;
    font-size: 20px;
    font-weight: bold;
    color: #34B289;
    border-radius: 3px;
    z-index: 9999;
    display: block;
    border: none;
    padding: 16px 32px;
    cursor: pointer;
    &:active {
        opacity: 0.8;
    }
`;

type DefaultProps = {}

type Props = {
    createTeam?: (name: string) => Promise<>,
    data: {
        loading: bool,
        teams: Array<Object>
    },
    router: {
        replace?: (path: Object) => void
    },
    location: {
        pathname: string
    }
}

type State = {
    activeTeam: ?Object,
    message: ?string
}

class Admin extends Component<DefaultProps, Props, State> {
    static defaultProps: DefaultProps = {};
    state: State = {
        activeTeam: null,
        message: null,
        router: {},
        data: {},
        pathname: {}
    }
    componentWillReceiveProps(nextProps: Props) {
        const { data: { teams, loading } } = nextProps;

        if (loading || !teams) return;
        this.setState({ activeTeam: teams[0] })


    }
    // createTeam = () => {
    //     const { teamName } = this.state;
    //     const { createTeam } = this.props;
    //     createTeam(teamName)
    //         .then(res => console.log(res))
    //         .catch(err => this.setState({ message: { type: 'error', message: 'shit' } }));
    // }
    selectTeam = (team: Object) => {
        this.setState({ activeTeam: team });
    }
    onContinue = () => {
        const {
            router: { replace },
            location: { pathname }
        } = this.props;
        const { activeTeam } = this.state;

        if (!activeTeam) return;
        selectTeam(activeTeam.id);
        if (replace) {
            replace({
                pathname: '/',
                state: { nextPathname: pathname }
            });
        }
    }
    renderTeams = () => {
        const { data } = this.props;
        if (data.loading) return <div>LOADING </div>;

        if (data.teams && data.teams.length > 0) {
            return (
                <div>
                    {data.teams.map(team =>
                        <Team
                            team={team}
                            onSelect={() => this.selectTeam(team)}
                            key={team.id}
                        />
                    )}
                </div>
            );
        }

        return <div>You dont have any teams yet</div>
    }
    render() {
        const { activeTeam } = this.state;
        return (
            <div className={css.container}>
                <div className={css.SidePanel}>
                    <img src={logo} alt="presentation" />
                    <div className={css.SidePanelInner}>
                        <h1> Hi there! Welcome to Cuest.</h1>
                        <p> We've created a team for you, so all you need to do is Sign in. </p>
                    </div>
                </div>
                {/* <CreateTeam
                     onChange={value => this.setState({ teamName: value })}
                     onSubmit={this.createTeam}
                     teamName={teamName}
                /> */}
                <div className={css.teamList}>
                    <div className={css.teamListInner}>
                        {this.renderTeams()}
                    </div>
                    <Submit onClick={this.onContinue}>
                        <span>Sign in to {activeTeam && activeTeam.name}</span>
                    </Submit>
                </div>
            </div>
        );
    }
}


export default Admin;
