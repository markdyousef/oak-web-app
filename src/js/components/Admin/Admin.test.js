import React from 'react';
import { shallow } from 'enzyme';
import Admin from './Admin';

it('renders correctly', () => {
    const props = {
        createTeam: () => {},
        data: {}
    };
    const shallowComponent = shallow(<Admin {...props} />);
    expect(shallowComponent).toMatchSnapshot();
});
