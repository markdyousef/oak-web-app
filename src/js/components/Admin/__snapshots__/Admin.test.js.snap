exports[`test renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className={undefined}>
    <CreateTeam
        onChange={[Function]}
        onSubmit={[Function]}
        teamName="" />
</div>,
  "nodes": Array [
    <div
      className={undefined}>
      <CreateTeam
            onChange={[Function]}
            onSubmit={[Function]}
            teamName="" />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Admin
        createTeam={[Function]} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Admin {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "createTeam": [Function],
        "props": Object {
          "createTeam": [Function],
        },
        "refs": Object {},
        "state": Object {
          "teamName": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className={undefined}>
          <CreateTeam
                    onChange={[Function]}
                    onSubmit={[Function]}
                    teamName="" />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className={undefined}>
          <CreateTeam
                    onChange={[Function]}
                    onSubmit={[Function]}
                    teamName="" />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Admin
    createTeam={[Function]} />,
}
`;
